hack/update-vendor.sh

```diff
@@ -0,0 +1,415 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+ORIGIN_ROOT=$(cd $(dirname "${BASH_SOURCE[0]}")/..; pwd)
+source "${ORIGIN_ROOT}/hack/lib/init.sh"
+
+
+function kube::log::status() {
+    echo "$@"
+}
+
+
+function kube::log::error() {
+    echo "$@" 1>&2
+}
+
+# Explicitly opt into go modules, even though we're inside a GOPATH directory
+export GO111MODULE=on
+# Explicitly clear GOFLAGS, since GOFLAGS=-mod=vendor breaks dependency resolution while rebuilding vendor
+export GOFLAGS=
+# Ensure sort order doesn't depend on locale
+export LANG=C
+export LC_ALL=C
+# Detect problematic GOPROXY settings that prevent lookup of dependencies
+if [[ "${GOPROXY:-}" == "off" ]]; then
+  kube::log::error "Cannot run hack/update-vendor.sh with \$GOPROXY=off"
+  exit 1
+fi
+
+# TODO: kube::golang::verify_go_version
+# TODO: kube::util::require-jq
+
+TMP_DIR="${TMP_DIR:-$(mktemp -d /tmp/update-vendor.XXXX)}"
+LOG_FILE="${LOG_FILE:-${TMP_DIR}/update-vendor.log}"
+kube::log::status "logfile at ${LOG_FILE}"
+
+if [ -z "${BASH_XTRACEFD:-}" ]; then
```

Comment:
this, this is why people think you're weird in bash.  What does this do?

Reply:
don't know, ask @liggitt ;-)

Reply:
It dumps all invocations to a file. Was handy for debugging.